### PR TITLE
fix: (core) Datepicker and DateTimePicker: Respect disabled/block function by inputs

### PR DESCRIPTION
**Pull request by JKMarkowski**
#### Please provide a link to the associated issue.
fixes: https://github.com/SAP/fundamental-ngx/issues/1627
#### Please provide a brief summary of this pull request.
There is special behaviour for `Date Picker` and `Datetime Picker`, which doesn't allow to pick disabled or blocked date by writing it in input. Before it was completely allowed.
I also tried to follow behaviour from https://sapui5.hana.ondemand.com/#/entity/sap.m.DatePicker/sample/sap.m.sample.DatePicker 

I also removed useless inputs form DateTimePicker
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/date-picker/date-picker-docs.component.html [hunk 1]
```diff
@@ -74,7 +74,9 @@
 <fd-docs-section-title [id]="'combobox-reactiveForm'" [componentName]="'datePicker'">
     Date Picker in Reactive Form
 </fd-docs-section-title>
-<description>The date-picker component may also be used within Angular Reactive Forms.</description>
+<description>The date-picker component may also be used within Angular Reactive Forms. There is used
+    <code>[disableFunction]</code>, which controls the <code>valid</code> flag
```
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/date-picker/date-picker-docs.component.html]: Maybe 
"<code>[disableFunction]</code> is used to control the <code>valid</code> flag"

**Comment by InnaAtanasova** [comment]
LGTM (I will not approve now to give a chance to the others to also take a look at it)

**Comment by InnaAtanasova** [comment]
@mpienkowski could you take a look at the fix for the issue you opened? Thank you!

File: apps/docs/src/app/core/component-docs/date-picker/date-picker-docs.component.html (hunk 1 above)
Comment (stefanoScalzo) [reply apps/docs/src/app/core/component-docs/date-picker/date-picker-docs.component.html]: the `[disableFunction]` is controls the valid flag

File: libs/core/src/lib/date-picker/date-picker.component.ts [hunk 2]
```diff
@@ -417,21 +420,35 @@ export class DatePickerComponent implements ControlValueAccessor, Validator {
...
+                }
+
+                this.isInvalidDateInput = !this.isRangeModelValid(selectedRangeDate);
+
+                /** If start date is invalid, cause of format, block or disable function, there is invalidDate forced */
+                if (!this.isStartDateValid(selectedRangeDate.start)) {
+                    selectedRangeDate.start = this.invalidDate();
+                }
+
+                /** If end date is invalid, cause of format, block or disable function, there is invalidDate forced */
```
Comment (stefanoScalzo) [inline libs/core/src/lib/date-picker/date-picker.component.ts]: do you mean because "because of format, block..."?

**Comment by stefanoScalzo** [comment]
Is there any example of showing a disabled range of dates?

File: libs/core/src/lib/date-picker/date-picker.component.ts (hunk 2 above)
Comment (JKMarkowski) [reply libs/core/src/lib/date-picker/date-picker.component.ts]: It's changed to `because`

**Comment by JKMarkowski** [comment]
Hi @stefanoScalzo, there is example of disabledFunction on range mode.

**Comment by salarenko** [comment]
> Hi @stefanoScalzo, there is example of disabledFunction on range mode.

I thinks we should create separate example focusing only disabling dates. It took me a while until I found the one using `disableFunction` even with your hint.
We have two examples of datepicker with reactive forms:
-  Date Picker in Reactive Form
-  Date Picker in Reactive Form with a Range of Dates.
Maybe lets leave one more complex example of reactive forms and transform the other one into "Disabling datepicker and custom dates" section.

File: libs/core/src/lib/datetime-picker/datetime-picker.component.ts [hunk 3]
```diff
@@ -453,7 +417,15 @@ export class DatetimePickerComponent implements OnInit, OnDestroy, ControlValueA
     private isModelValid(fdDateTime: FdDatetime): boolean {
         return fdDateTime &&
             fdDateTime instanceof FdDatetime &&
-            fdDateTime.isDateValid() && fdDateTime.isTimeValid();
+            this.isDateValid(fdDateTime) && fdDateTime.isTimeValid();
```
Comment (salarenko) [inline libs/core/src/lib/datetime-picker/datetime-picker.component.ts]: New line
```suggestion
            this.isDateValid(fdDateTime) && 
            fdDateTime.isTimeValid();
```

**Comment by JKMarkowski** [comment]
@salarenko All of the changes are included, I added `_` prefix to private fields and methods on date/datetime picker component. 
Also there is new example of range disable date picker.

**Comment by stefanoScalzo** [comment]
> > Hi @stefanoScalzo, there is example of disabledFunction on range mode.
> 
> I think we should create separate example focusing only disabling dates. It took me a while until I found the one using `disableFunction` even with your hint.
> We have two examples of datepicker with reactive forms:
> 
> * Date Picker in Reactive Form
> * Date Picker in Reactive Form with a Range of Dates.
> 
> Maybe lets leave one more complex example of reactive forms and transform the other one into "Disabling datepicker and custom dates" section.

I couldn't agree more! We need to clarify to the users which example is for what in a quick glance

**Pull request merged**